Configurations/10-main.conf

```diff
@@ -140,6 +140,19 @@ sub vms_info {
                 $vms_info->{def_zlib} =~ s|/.*$||g;
             }
         }
+
+        if ($config{target} =~ /-ia64/) {
+            `PIPE ias -H 2> NL:`;
+            if ($? == 0) {
+                $vms_info->{as} = "ias";
+                $vms_info->{asflags} = '-d debug "-N" vms_upcase';
+                $vms_info->{asflags} .= ' "-Milp32"'
+                    if $config{target} =~ /-p32$/;
+                $vms_info->{asflags} .= ' "-Mp64"'
+                    if $config{target} =~ /-p64$/;
```

Comment:
On side note. From what you've shown me, it doesn't seem to matter which -M option you pass. As far as I understand binary object format is the same in both cases, which makes p32 vs. p64 really just matter of application code convention(*). And thing is that current assembly modules don't do that kind of pointer arithmetic. Since it's unlike that there will be new IA64 modules, one can just as well *simplify* this by omitting the -M flag altogether and letting it be using the default setting in all cases.

(*) Just like on 64-bit Windows. However, unlike VMS Windows don't have 32-bit "system" headers. If making analogies with VMS, it would be equivalent of #pragma pointer_size 64 in beginning of each "system" header and #pragma pointer_size restore at the end.

Reply:
Right, I forgot to edit them away...  and it makes sense as long as we code assembler as precisely as we currently do.